test/Platforms/Echo.Platforms.AsmResolver.Tests/Emulation/Dispatch/ObjectModel/CastHandlerTest.cs

```diff
@@ -179,5 +181,77 @@ public void CastMatchedRefTypeShouldPushSameObject(CilCode code, Type t1, Type t
             var value = Assert.Single(stack);
             Assert.Equal(address, value.Contents.AsSpan().ReadNativeInteger(Context.Machine.Is32Bit));
         }
+
+        [Fact]
+        public void CastGenericTypeReturnsInt16() {
+            var vm = Context.Machine;
+
+            var genericMethod = ModuleFixture.MockModule.TopLevelTypes
+                .First(t => t.Name == nameof(SimpleClass))
+                .Methods
+                .First(m => m.Name == nameof(SimpleClass.GenericMethod));
+
+            var methodSpecification = new MethodSpecification(genericMethod,
+                new GenericInstanceMethodSignature(ModuleFixture.MockModule.CorLibTypeFactory.Int16));
+
+            short value = 32000;
+
+            var returnValue = vm.Call(methodSpecification, new object[] { value });
```

Comment:
I am not sure if we want to use `vm.Call` here. This is a pretty complicated function, which means we are implicitly testing a lot of different components of the VM with this single unit test. I think it may be better to use the `Dispatcher.Dispatch` method to emulate a single instruction (as is done with the other tests) so that if the test fails in the future, we know exactly that it was because of that particular instruction, and not have to guess which hidden component of the VM may be the problem here.